### PR TITLE
cmd/geth: rename --whitelist to --eth.requiredblocks

### PR DESCRIPTION
**Pull request by ryanschneider**
For awhile now I've wanted to rename the "whitelist" option I introduced in https://github.com/ethereum/go-ethereum/pull/18028 to something both more descriptive (the current flag does not convey _what_ is being whitelisted) and more inclusive.

IMO `--peer.requiredblocks` is much more self-documenting, though I'm totally open to having a discussion around other argument names (and will make the required changes to this PR if we decide on a different name).

### CODE REVIEW

**Comment by fjl** [comment]
`requiredblocks` is a very good name. I don't really agree with the `peer.` prefix because we don't have any other flag like that.

**Comment by ryanschneider** [comment]
I agree that the peer prefix is new, but feel like requiredblocks by itself
still suffers some of the same issues as the old name: it’s still unclear
which subsystem requires these blocks from who. peerrequiredblocks (no dot
namespace) is very hard to parse especially with the double r.

I’m willing to move forward with requiredblocks though if others feel it’s
clear enough.

**Comment by holiman** [comment]
I think it's good to have `peer` in there -- because the current implementation just 'shields' it on the peer-level, not in any deeper level. E.g. if a peer lies in the initial checks, it can still serve us those blocks. 
However, one could argue that the peer-part is an implementation detail, and the 'intention' is to require certain hashes.

**Comment by holiman** [comment]
Yeah I'm fine with skipping the `peer.` prefix

**Comment by fjl** [comment]
We have now settled on `--eth.requiredblocks`.
